File: yeonkyung/주식가격.py
def solution(prices):
    answer = [0] * len(prices)
    # 기간 차이를 계산해야 하므로 인덱스를 추가
    stack = []

    for idx, price in enumerate(prices):
        while stack:
            if prices[stack[-1]] > price:
                last = stack.pop()
                answer[last] = idx - last
            else:
                break
        stack.append(idx)

    while stack:
        last = stack.pop()
        answer[last] = idx - last

    return answer

if __name__ == "__main__":
    prices = [1, 2, 3, 2, 3]
    print(solution(prices))

File: Doodream/주식가격.java
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = {};
        // answer의 크기는 prices의 크기와 같다.
        answer = new int[prices.length];
        // 가격이 내려가지 않는 시간.
        int count  = 0;
        // prices[0]부터 시작해서 price[prices.length - 1]까지 간다.
        for (int i = 0; i < prices.length; i++) {
            // prices[i]의 비교는 price[i + 1]부터 시작되므로
            for (int j = i + 1; j < prices.length; j++) {
                // prices[i]가 prices[j]보다 작다면 count ++
                if(prices[i] <= prices[j]){
                    count++;
                }
                // prices[i]가 price[j] 보다 크다면 count ++후 종료 ,answer[i+1]차례로 넘어감
                else{
                    count++;
                    break;
                }

            }
            //answer[i]는 count를 넣고
            answer[i] = count;
            //count는 다음 answer[i + 1]를 위해서 초기화.
            count = 0;
        }
        return answer;
    }
}